### ISSUE TITLE
Discrepancy tokensregex webserver and CoreNLP in Python?

### ISSUE THREAD

**Issue by rickbeeloo** [stanfordnlp/CoreNLP#1075]
Let's take the following setence: `organic wastes under variable temperature conditions` and pattern: `[{tag:/JJ/}]*[{tag:/NN.*/}]+`
When we pass this to http://corenlp.run/:
![image](https://user-images.githubusercontent.com/19516376/88464610-86440680-cebc-11ea-98dd-493fa6087b49.png)

Then when we do this in Python:
```
with CoreNLPClient(memory='16G', threads=1, annotators=['tokenize','ssplit','pos','lemma','ner','depparse']) as client:
    text = 'organic wastes under variable temperature conditions'
    print(client.tokensregex(text, '[{tag:/JJ/}]*[{tag:/NN.*/}]+'))
```
It will print: `{'sentences': [{'0': {'text': 'organic wastes', 'begin': 0, 'end': 2}, '1': {'text': 'variable temperature', 'begin': 3, 'end': 5}, '2': {'text': 'conditions', 'begin': 5, 'end': 6}, 'length': 3}]}` Note that the webserver finds "variable temperature conditions" whereas in Python we only find "variable temperature" and "conditions" as seperate matches. I need the same output as the webserver

**Comment by rickbeeloo** [comment]
Upon inspection of the request header for the webserver I noticed it adds a a `\` before the last `+`, so the pattern should be '[{tag:/JJ/}]*[{tag:/NN.*/}]\+'))

**Comment by AngledLuffa** [comment]
I'm not sure this deserves to be closed.  Didn't you have the expectation that the API does the escaping for you?  Perhaps we should fix that in the stanza client.

**Comment by rickbeeloo** [comment]
Yes I indeed expected to just be able to copy a regex from corenlp.run to obtain the same results (and thus also the same parsing)

**Comment by rickbeeloo** [comment]
I also noticed that entering the escaped regex, thus `[{tag:/JJ/}]*[{tag:/NN.*/}]\+` on the werbserver will throw `java.lang.RuntimeException: Error when parsing [{tag:/JJ/}]*[{tag:/NN.*/}]\+` this makes it even harder to test a regex and then use it for e.g. Python

**Comment by AngledLuffa** [comment]
I thought at first this was on the stanza side, but then I discovered it
was an issue with the java code.  The result is that although I fixed it,
the fix didn't make it into the 4.1.0 version currently being built.  It
will be available in the next release or on github, though.

**Comment by rickbeeloo** [comment]
Aaah awesome!

**Comment by AngledLuffa** [comment]
https://github.com/stanfordnlp/CoreNLP/commit/5e54ae4862d38de1e36020380b9dab48ab73eebc
